lib/rules/use-object-in-propequal.js

```diff
@@ -0,0 +1,112 @@
+ * @author Kevin Partington
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const assert = require("assert"),
+    utils = require("../utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        type: "problem",
+        docs: {
+            description: "enforce use of objects as expected value in `assert.propEqual`",
+            category: "Possible Errors",
+            url: "https://github.com/platinumazure/eslint-plugin-qunit/blob/master/docs/rules/use-object-in-propequal.md"
+        },
+        messages: {
+            useObject: "Use object as propEqual expected value (found: {{ value }})."
+        },
+        schema: []
+    },
+
+    create: function (context) {
+        // Declare a test stack in case of nested test cases (not currently supported by QUnit).
+        const sourceCode = context.getSourceCode(),
+            testStack = [];
+
+        function getCurrentAssertContextVariable() {
+            assert(testStack.length, "Test stack should not be empty");
+
+            return testStack[testStack.length - 1].assertVar;
+        }
+
+        function isGlobalPropEqual(calleeNode) {
+            return calleeNode &&
+                calleeNode.type === "Identifier" &&
+                calleeNode.name === "propEqual";
+        }
+
+        function isAssertPropEqual(calleeNode) {
+            return calleeNode &&
+                calleeNode.type === "MemberExpression" &&
+                calleeNode.property.type === "Identifier" &&
+                calleeNode.property.name === "propEqual" &&
+                calleeNode.object.type === "Identifier" &&
+                calleeNode.object.name === getCurrentAssertContextVariable();
+        }
+
+        function isPropEqual(calleeNode) {
+            return isAssertPropEqual(calleeNode) || isGlobalPropEqual(calleeNode);
+        }
+
+        function isNonObject(argNode) {
+            switch (argNode.type) {
+                case "ObjectExpression":
+                case "ArrayExpression":
+                case "ThisExpression":
+                case "Identifier":
+                case "MemberExpression":
+                case "CallExpression":
+                case "NewExpression":
+                    return false;
+
+                case "Literal":
+                default:
+                    return true;
+            }
+        }
+
+        function hasNonObjectExpectedValue(callExpressionNode) {
+            return callExpressionNode &&
+                callExpressionNode.arguments &&
+                callExpressionNode.arguments.length >= 2 &&
+                isNonObject(callExpressionNode.arguments[1]);
```

Comment:
Instead of using this `isNonObject`, can't we just check `node.type === 'Literal'`? It seems like literals are the only thing that we want to disallow here.

And I'm concerned about the `isNonObject` function because:
* it has a negative in the name which is confusing
* it has a list of node types which could easily be incomplete (what about OptionalCallExpression, OptionalMemberExpression, etc).

Reply:
I've pushed some commits. When you have an opportunity, please take a look at the latest commit and let me know what you think.